internal/types/path.go

```diff
@@ -17,17 +17,68 @@ package types
 import (
 	"fmt"
 	"strconv"
+	"strings"
 )
 
+// Path represents field path type. It should be used wherever we work with paths or dot notation.
+type Path struct {
+	s []string
+}
+
+// Len returns path length.
+func (p Path) Len() int {
+	return len(p.s)
+}
+
+// Slice returns path slice copy.
+func (p Path) Slice() []string {
+	path := make([]string, len(p.s))
+	copy(path, p.s)
+	return path
+}
+
+// Element returns path element specified by the index or empty string if index below 0 or greater than Path len.
```

Comment:
I'd suggest dividing this sentence into two. One might misread the "specified by the index or empty string" part :)

Maybe something like
```
Element returns path's element specified by the index.
If index is below 0 or greater than Path length, it returns an empty string.
```


P.S. I actually like how the doc for `Remove` looks like! Maybe something similar here?